CRM/Contactlayout/BAO/ContactLayout.php

```diff
@@ -35,15 +35,38 @@ public static function getLayout($cid, $uid = NULL) {
     }
     $get->addClause('OR', $subClauses);
 
+    // Loop over candidate layouts to return the first match
     foreach ($get->execute() as $layout) {
-      if (self::checkGroupFilter($uid, $layout)) {
+      if (self::checkSubtypeFilter($contact, $layout) && self::checkGroupFilter($uid, $layout)) {
         self::loadBlocks($layout, $contact['contact_type']);
         return $layout;
       }
     }
     return NULL;
   }
 
+  /**
+   * Check if the user matches the sub_type filter for a layout
+   *
+   * @param array $contact
+   * @param array $layout
+   *
+   * @return bool
+   */
+  private static function checkSubtypeFilter($contact, $layout) {
+    // AND operator, check if the contact matches all layout sub_type(s)
+    if ($layout['settings']['sub_type_operator'] == 'AND') {
```

Comment:
This array key might not exist.

```suggestion
    if (($layout['settings']['sub_type_operator'] ?? 'OR') === 'AND') {
```